src/core/common/message.hpp

```diff
@@ -1225,6 +1227,45 @@ class Message : public otMessage, public Buffer
 #endif // #if OPENTHREAD_CONFIG_MULTI_RADIO
 
 protected:
+    class ConstIterator : public ItemPtrIterator<const Message, ConstIterator>
+    {
+        friend class ItemPtrIterator<const Message, ConstIterator>;
+
+    public:
+        ConstIterator(void) = default;
+
+        explicit ConstIterator(const Message *aMessage)
+            : ItemPtrIterator(aMessage)
+        {
+        }
+
+    private:
+        void Advance(void) { mItem = (mItem != nullptr) ? mItem->GetNext() : nullptr; }
```

Comment:
```suggestion
        void Advance(void) { mItem = mItem->GetNext(); }
```

Should `Advance()` never be called when mItem is nullptr ?

Reply:
Yes. I think we can remove this. 
If `Iterator` is at end, calling `++` or `Advance()` on it is usually undefined behavior (incorrect use).

Reply:
Changed this in new push.